internal/apis/certmanager/validation/certificate.go

```diff
@@ -118,6 +123,26 @@ func ValidateCertificateSpec(crt *internalcmapi.CertificateSpec, fldPath *field.
 		el = append(el, validateEmailAddresses(crt, fldPath)...)
 	}
 
+	if len(crt.OtherNames) > 0 {
+		if !utilfeature.DefaultFeatureGate.Enabled(feature.OtherNames) {
+			el = append(el, field.Forbidden(fldPath.Child("OtherNames"), "Feature gate OtherNames must be enabled on both webhook and controller to use the alpha `otherNames` field"))
+		}
```

Comment:
suggestion: Is this missing a return or an else? As written, we'll still parse the otherNames if the feature gate is disabled. I'd expect we should ignore them entirely - what do you think?